packages/template-retail-react-app/app/components/unavailable-product-confirmation-modal/index.jsx

```diff
@@ -14,28 +14,52 @@ import {useDisclosure} from '@salesforce/retail-react-app/app/components/shared/
 import {noop} from '@salesforce/retail-react-app/app/utils/utils'
 
 /**
- * This Component will responsible to determine of a given product ids has become unavailable
- * and will prompt the users to remove them before proceeding any further
+ * This Component determines if the provided products have become unavailable or out of stock or low stock that
+ * can't be fulfilled and will prompt the users to remove them before proceeding any further
  *
  * @param productIds -  list of product ids to check for availability
+ * @param productItems -  basket product items
  * @param handleUnavailableProducts - callback function to handle what to do with unavailable products
  * @returns {JSX.Element} -  Conformation Modal Component
  *
  */
 const UnavailableProductConfirmationModal = ({
     productIds = [],
+    productItems = [],
     handleUnavailableProducts = noop
 }) => {
     const unavailableProductIdsRef = useRef(null)
+    const ids = productIds.length ? productIds : productItems.map((i) => i.productId)
```

Comment:
I see this is the main change from the previous PR (#1865) 

So now we can either set productIds (the previous API) or productItems (the new API). I think this makes sense to prioritize the current behavior so existing projects will opt in to transition to the new API rather than having the new API forced on them.